src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsProvider.cs

```diff
@@ -354,11 +355,23 @@ protected void MergeExistingInMemoryGlobalSettings(LaunchSettingsData newSnapsho
         }
 
         /// <summary>
-        /// Helper function to set the new snapshot and post the change to consumers.
+        /// Helper function to set the new snapshot, update the active profile property to match what is in the snapshot,  and post
+        /// the changes to consumers.
         /// </summary>
-        protected void FinishUpdate(ILaunchSettings newSnapshot)
+        protected async Task FinishUpdateAsync(ILaunchSettings newSnapshot, bool ensureProfileProperty)
         {
             CurrentSnapshot = newSnapshot;
+            if(ensureProfileProperty)
+            {
+                var props = await CommonProjectServices.ActiveConfiguredProjectProperties.GetProjectDebuggerPropertiesAsync().ConfigureAwait(true);
+                if (await props.ActiveDebugProfile.GetValueAsync().ConfigureAwait(true) is IEnumValue activeProfileVal)
```

Comment:
Is it deliberate that this ConfigureAwait(true)?

Reply:
I would expect it to be false. I copied that code form another place in the code and it is true there too. I see about 5 of these. I'll fix those too and re-push. Thanks for catching that.